<a href="https://colab.research.google.com/github/shwetaNayak19/Bank-Marketing-Effectiveness-Prediction/blob/main/Bank_Marketing_Effectiveness_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Problem Description </b>

### The data is related with direct marketing campaigns (phone calls) of a Portuguese banking institution. The marketing campaigns were based on phone calls. Often, more than one contact to the same client was required, in order to access if the product (bank term deposit) would be ('yes') or not ('no') subscribed. The classification goal is to predict if the client will subscribe a term deposit (variable y).


#**Importing required Libraries for analysis**.

In [ ]:
import pandas as pd # for data handling and manipulation
import numpy as np
import matplotlib.pyplot as plt # for data visualization
import seaborn as sns # for data visualization
# magic function to render graphs/charts in notebook
%matplotlib inline  
import plotly.graph_objects as go
import plotly.express as px
#import datetime
import warnings
warnings.filterwarnings('ignore')

In [ ]:
#set style for graph
#sns.set_style(style='darkgrid')

# Setting fontsizes,fontweight,label weight for labels,titles.
#plt.rcParams["font.weight"] = "bold"
#plt.rcParams["axes.labelweight"] = "bold"
#plt.rcParams["axes.titlesize"] = 25
#plt.rcParams["axes.titleweight"] = 'bold'
#plt.rcParams['xtick.labelsize']=15
#plt.rcParams['ytick.labelsize']=15
#plt.rcParams["axes.labelsize"] = 20
#plt.rcParams["legend.fontsize"] = 15
#plt.rcParams["legend.title_fontsize"] = 15

#**Data Loading and Preperation**

#**Mounting Drive**

In [ ]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


#**Reading Data in Notebook**

In [ ]:
#reading csv file 
Bank_market_df = pd.read_csv('/content/drive/MyDrive/BankMarket_prediction/bank-full.csv', delimiter=';')

#**Exploring Dataset**

In [ ]:
#checking top 5 rows of dataset
Bank_market_df.head()

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no

In [ ]:
Bank_market_df.tail()

age           job   marital  education default  balance housing loan  \
45206   51    technician   married   tertiary      no      825      no   no   
45207   71       retired  divorced    primary      no     1729      no   no   
45208   72       retired   married  secondary      no     5715      no   no   
45209   57   blue-collar   married  secondary      no      668      no   no   
45210   37  entrepreneur   married  secondary      no     2971      no   no   

         contact  day month  duration  campaign  pdays  previous poutcome    y  
45206   cellular   17   nov       977         3     -1         0  unknown  yes  
45207   cellular   17   nov       456         2     -1         0  unknown  yes  
45208   cellular   17   nov      1127         5    184         3  success  yes  
45209  telephone   17   nov       508         4     -1         0  unknown   no  
45210   cellular   17   nov       361         2    188        11    other   no

In [ ]:
Bank_market_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        45211 non-null  int64 
 1   job        45211 non-null  object
 2   marital    45211 non-null  object
 3   education  45211 non-null  object
 4   default    45211 non-null  object
 5   balance    45211 non-null  int64 
 6   housing    45211 non-null  object
 7   loan       45211 non-null  object
 8   contact    45211 non-null  object
 9   day        45211 non-null  int64 
 10  month      45211 non-null  object
 11  duration   45211 non-null  int64 
 12  campaign   45211 non-null  int64 
 13  pdays      45211 non-null  int64 
 14  previous   45211 non-null  int64 
 15  poutcome   45211 non-null  object
 16  y          45211 non-null  object
dtypes: int64(7), object(10)
memory usage: 5.9+ MB


In [ ]:
Bank_market_df.shape

(45211, 17)

#**Inference.**
* Total No of entries- 45211.
* Total of columns- 17.

In [ ]:
Bank_market_df.describe().T

count         mean          std     min    25%    50%     75%  \
age       45211.0    40.936210    10.618762    18.0   33.0   39.0    48.0   
balance   45211.0  1362.272058  3044.765829 -8019.0   72.0  448.0  1428.0   
day       45211.0    15.806419     8.322476     1.0    8.0   16.0    21.0   
duration  45211.0   258.163080   257.527812     0.0  103.0  180.0   319.0   
campaign  45211.0     2.763841     3.098021     1.0    1.0    2.0     3.0   
pdays     45211.0    40.197828   100.128746    -1.0   -1.0   -1.0    -1.0   
previous  45211.0     0.580323     2.303441     0.0    0.0    0.0     0.0   

               max  
age           95.0  
balance   102127.0  
day           31.0  
duration    4918.0  
campaign      63.0  
pdays        871.0  
previous     275.0

In [ ]:
#check the column names 
Bank_market_df.columns

Index(['age', 'job', 'marital', 'education', 'default', 'balance', 'housing',
       'loan', 'contact', 'day', 'month', 'duration', 'campaign', 'pdays',
       'previous', 'poutcome', 'y'],
      dtype='object')

#**Data Description**

**Input variables:**
**Bank Client data:**

**age**: (numeric): age in years.

**job :** type of job (categorical): ['admin' , 'blue-collar' ,'entrepreneur' , 'housemaid' , 'management', 'retired' , 'self-employed' ,'services' , 'student' , 'technician' , 'unemployed' , 'unknown' ]

**marital :** marital status (categorical): 'divorced' , 'married' ,'single' , 'unknown' ; note: 'divorced' means divorced or widowed.

**education :**(categorical): ['tertiary' , 'secondary', 'primary' , 'unknown']

**default:** has credit in default? (categorical): 'no', 'yes', 'unknown'

**housing:** has housing loan? (categorical): 'no', 'yes', 'unknown'

**loan:** has personal loan? (categorical): 'no', 'yes', 'unknown')
Related with the last contact of the current campaign:

**contact:** contact communication type (categorical): 'cellular',   'telephone'.

**month:** last contact month of year (categorical): 'jan', 'feb', 'mar', ..., 'nov', 'dec'

**day_of_week:** last contact day of the week (categorical): 'mon','tue','wed', 'thu', 'fri'.

**duration:** last contact duration, in seconds (Numeric). 

Important note: this attribute highly affects the output target (e.g., if duration=0 then y='no'). Yet, the duration is not known before a call is performed. Also, after the end of the call y is obviously known. Thus, this input should only be included for benchmark purposes and should be discarded if the intention is to have a realistic predictive model.

**Other attributes:**

**campaign:**number of contacts performed during this campaign and for this client (numeric, includes last contact)

**pdays:** number of days that passed by after the client was last contacted from a previous campaign (numeric; 999 means client was not previously contacted)

**previous:** number of contacts performed before this campaign and for this client (numeric)

**poutcome:** outcome of the previous marketing campaign (categorical):  'failure', 'nonexistent', 'success')

**Output variable** (desired target):

**y** - has the client subscribed a term deposit? (binary:  'yes' , 'no')

In [ ]:
Bank_market_df.dtypes

age           int64
job          object
marital      object
education    object
default      object
balance       int64
housing      object
loan         object
contact      object
day           int64
month        object
duration      int64
campaign      int64
pdays         int64
previous      int64
poutcome     object
y            object
dtype: object